peerdid/peer_did_utils.py

```diff
@@ -0,0 +1,266 @@
+import base64
+import hashlib
+import json
+import re
+from typing import Union, List
+
+import base58
+import varint
+
+from peerdid.types import JSON, PublicKeyAgreement, PublicKeyAuthentication, PublicKeyTypeAgreement, \
+    PublicKeyTypeAuthentication, PEER_DID, EncodingType
+
+
+def _encode_service(service: JSON) -> str:
+    """
+    Generates encoded service according to the second algorithm
+    (https://identity.foundation/peer-did-method-spec/index.html#generation-method)
+    For this type of algorithm did_doc can be obtained from peer_did
+
+    :param service: JSON string conforming to the DID specification (https://www.w3.org/TR/did-core/#services)
+
+    :return: encoded service
+    """
+    service_to_encode = re.sub(r"[\n\t\s]*", "", service) \
+        .replace('\'', '\"') \
+        .replace("type", "t") \
+        .replace("serviceEndpoint", "s") \
+        .replace("didcommmessaging", 'dm') \
+        .replace("routingKeys", 'r') \
+        .encode('ascii')
+    return '.S' + base64.b64encode(service_to_encode).decode("utf-8")
```

Comment:
does this imply that non-utf8 `service_to_encode`s will become utf-8 after this transformation?

Reply:
UTF-8 is backward-compatible with ASCII, but i think it will be better to change `encode('ascii')` to `encode('utf-8')`